### PR TITLE
pkg/ansible - Bugfix/update owner ref behavior

### PR DESCRIPTION
**Pull request by shawn-hurley**
**Description of the change:**
* Ansible operator would create resources with owner references that were incorrect.
* This uses a new mechanism to dynamically watch resources that cross the namespace boundaries.

**Motivation for the change:**
* This was a very bad assumption on our part and is causing issues.

### CODE REVIEW

**Comment by shawn-hurley** [comment]
@jmrodri Please take a look when you get a chance.

File: pkg/ansible/controller/reconcile.go [hunk 1]
```diff
@@ -67,6 +67,7 @@ func (r *AnsibleOperatorReconciler) Reconcile(request reconcile.Request) (reconc
 		return reconcile.Result{}, nil
 	}
 	if err != nil {
+		log.Error(err, "hhere", "namespace_name", request.NamespacedName)
```
Comment (jmrodri) [inline pkg/ansible/controller/reconcile.go]: is "hhere" intended? or a typo?

File: pkg/ansible/proxy/proxy.go [hunk 2]
```diff
@@ -161,16 +161,30 @@ func CacheResponseHandler(h http.Handler, informerCache cache.Cache, restMapper
...
 							if oRef.APIVersion == ownerRef.APIVersion && oRef.Kind == ownerRef.Kind {
-								err := addWatchToController(ownerRef, cMap, un, restMapper)
+								err := addWatchToController(ownerRef, cMap, un, restMapper, true)
 								if err != nil {
 									log.Error(err, "Could not recover dependent resource watch", "owner", ownerRef)
 									return
 								}
 							}
 						}
+						if typeString, ok := un.GetAnnotations()[osdkHandler.TypeAnnotation]; ok {
```
Comment (jmrodri) [inline pkg/ansible/proxy/proxy.go]: Wow how deeply nested is this code?

File: pkg/ansible/proxy/proxy.go [hunk 3]
```diff
@@ -284,6 +324,39 @@ func InjectOwnerReferenceHandler(h http.Handler, cMap *controllermap.ControllerM
...
+		log.Error(err, m)
+		return false, err
+
+	}
+	// We need to determine whether or not the owner is a cluster-scoped
+	// resource because enqueue based on an owner reference does not work if
+	// a namespaced resource owns a cluster-scoped resource
+	ownerGV, err := schema.ParseGroupVersion(owner.APIVersion)
+	if err != nil {
+		m := fmt.Sprintf("could not get broup version for: %v", owner)
```
Comment (jmrodri) [inline pkg/ansible/proxy/proxy.go]: typo: "broup"?

File: pkg/handler/enqueue_annotation.go [hunk 4]
```diff
@@ -0,0 +1,101 @@
+package handler
```
Comment (jmrodri) [inline pkg/handler/enqueue_annotation.go]: I had to add the header to the new file I added in my PR. I think you might hit that as well.

Comment (jmrodri) [reply pkg/handler/enqueue_annotation.go]:
```Checking for license header...
license header checking failed:
  /home/travis/gopath/src/github.com/operator-framework/operator-sdk/pkg/handler/enqueue_annotation.go
make: *** [test/sanity] Error 255``

**Comment by shawn-hurley** [comment]
@fabianvf @jmrodri Can I get another review when you guys get a chance?

File: pkg/ansible/proxy/proxy.go [hunk 5]
```diff
@@ -440,34 +509,26 @@ func addWatchToController(owner kubeconfig.NamespacedOwnerReference, cMap *contr
 		// Store watch in map
 		wMap.Store(resource.GroupVersionKind())
 		// Use EnqueueRequestForOwner unless user has configured watching cluster scoped resources and we have to
```
Comment (fabianvf) [inline pkg/ansible/proxy/proxy.go]: I know you didn't add this, but have to what?

**Comment by jmazzitelli** [comment]
@shawn-hurley I do not know if its something that I am doing wrong that this PR simply catches now, or if there is still a problem... please check out issue #1330 and comment in there your thoughts. Thanks.

**Pull request merged**